### DISCUSSION TITLE
Extract interaction points and corresponding objects in the ray tracer

### DISCUSSION THREAD

**Discussion by member67** [NVlabs/sionna#329] [Q&A] [answered]
Dear Sionna Team,

I use the Sionna ray tracer (RT) and now try to find the actual coordinates of points where the radio signal interacts with a physical object, i.e., points of scattering and specular reflections.

Is there a way to do that?

I have not really found anythig in this regards in the documentation or discussions. However, I saw in the code that a "paths" object (sionna.rt.Paths) as returned from compute_paths(), for example, has the two instance variables "_vertices" and "_objects".

The variable "_vertices" looks suspiciously like what I need (coordinates of points of interaction) - can anybody confirm that?

In addition, the objects the radio signals interact with would be interesting as well. Again, I suspect that the "_objects" instance variable of the Paths class returns unique indices of the objects the signal interacts with along the different propagation paths. I am not at all sure, however. 

Can anybody clarify?

Thank you very much for your help and contribution!

**Comment by faycalaa** [comment, marked as answer]
Hi,

[Paths.vertices](https://github.com/NVlabs/sionna/blob/7eb04f43356333386b4361361188e3810620a0d8/sionna/rt/paths.py#L675) provides the coordinates of the intersection points of the rays with the scene.
[Paths.objects](https://github.com/NVlabs/sionna/blob/7eb04f43356333386b4361361188e3810620a0d8/sionna/rt/paths.py#L687) provides the [SceneObject.object_id](https://github.com/NVlabs/sionna/blob/7eb04f43356333386b4361361188e3810620a0d8/sionna/rt/scene_object.py#L36) of the intersected objects.

Hope this helps

**Reply by member67** [reply]
It helps very much - thank you for the quick clarification!

**Reply by itsghaniz** [reply]
Hi, I am stuck at a similar problem. I am able to get the interaction material and vertices of each path. But I have seen that there are interaction coordinates for the line of sight path as well. How is this possible. Could anyone please explain this?
Thanks in advance

**Reply by faycalaa** [reply]
Hi @itsghaniz,

Because Sionna returns dense tensors to store path data (tensors in which all slices along any given dimension have the same length) the `Paths.vertices` tensors contain entries for all paths of any depth, including the line-of-sight. However, if the corresponding entry in `Paths.objects` is set to `-1`, then the intersection point should be ignored. Specifically, in the case of the line-of-sight, all entries in `Paths.objects` should be equal to `-1`, indicating that no intersection with the scene occurred.

**Reply by itsghaniz** [reply]
Thanks alot. Really appreciate the answer.

**Reply by itsghaniz** [reply]
Hi, hope you are well. There is one thing I am confused about. When I look into the properties of a path, it suggests a path is either made by a certain number of reflections or a certain number of defractions. My question is, that it is also possible that a path be made by lets say first reflection then diffraction and then the antenna in a single path. Does sionna do it ? Thanks in advance.

**Reply by faycalaa** [reply]
Hi @itsghaniz,

Sionna currently supports the following types of paths:
- Line-of-sight
- Specular reflections only (up to the order specified by `max_depth`)
- `N` specular reflections followed by a diffuse reflection (where `N + 1 <= max_depth`)
- First-order diffraction, i.e., Transmitter -> Diffraction -> Receiver

 Paths combining reflection and diffraction are not supported.

**Reply by itsghaniz** [reply]
Hi, thanks for the reply, really appreciate it. Another thing I have noticed is that for diffraction, I am able to get the vertices i.e. the point of interaction, but not the object_id for diffraction. Is there a way to get those ids as well?
